cmd/ctl/pkg/status/util/prefixWriter.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/kubectl/pkg/describe"
+	"k8s.io/kubectl/pkg/util/event"
+)
+
+// This file defines implementation of the PrefixWriter interface defined in "k8s.io/kubectl/pkg/describe"
+// This implementation is based on the one in the describe package, with a slight modification of having a baseLevel
+// on top of which any other indentations are added.
+// The purpose is be able to reuse functions in the describe package where the Level of the output is fixed,
+// e.g. DescribeEvents() only prints out at Level 0.
```

Comment:
Did you try `nestedPrefixWriter` ? It looks like it should allow you to write the events table, indented: https://github.com/kubernetes/kubernetes/blob/0b678bbb51a83e47df912f1205907418e354b281/staging/src/k8s.io/kubectl/pkg/describe/describe.go#L159

If not, then add a note to the top of this file explaining that parts of this code have been copied from a particular revision of Kubernetes and link to the file.

Reply:
Hmm. `nestedPrefixWriter` does not show up in the code I'm reading..
We still need the copied and slightly modified `DescribeEvents` because on this line they print the column headers by using `\n` and two spaces and not with Prefixing, that means that the column headers are slightly misformatted if we use the originial DescribeEvents to print Events for the CR.
I did remove the implementation of the PrefixWriter as it's not needed anymore.